govc vm.ls kills VIC appliance VM

So I've hit a really bizarre issue in `govc`/`govmomi`.
Steps to reproduce:

Install a VIC VCH on ESXi with `vic-machine` in the standard way, then, after updating your environment to point `docker` at your VCH:

`docker create --name foo busybox`
`cat some-archive.tar.gz | docker cp - foo:/`
`govc vm.ls $VCH_NAME /tmp`

`govc` will hang. The VCH will stop responding. `govc vm.ip` will hang. `govc vm.info` will show the endpoint VM as powered on, with no IP address. Prior to calling `govc vm.ls`, everything works normally in the VCH.

I was also able to procure a stack trace from the toolbox code in `vic-init`:
```Nov 22 2017 19:16:27.327Z DEBUG op=189.1: Running logrotate: /usr/sbin/logrotate /tmp/vic-logrotate-conf-825382244
Nov 22 2017 19:16:27.360Z DEBUG op=189.1: logrotate finished successfully
Nov 22 2017 19:16:27.363Z DEBUG Inspecting children with status change
Nov 22 2017 19:16:27.366Z DEBUG No more child processes to reap
panic: runtime error: slice bounds out of range

goroutine 10 [running]:
github.com/vmware/vic/vendor/github.com/vmware/govmomi/toolbox.(*CommandServer).ListFiles(0xc420258cc0, 0x5d00d0001, 0x3300000062, 0xd00000022, 0xb10000000d, 0xffffffff00000000, 0xb1, 0x100000000, 0xc4203a564c, 0x2f, ...)
    /go/src/github.com/vmware/vic/vendor/github.com/vmware/govmomi/toolbox/command.go:515 +0x442
github.com/vmware/vic/vendor/github.com/vmware/govmomi/toolbox.(*CommandServer).ListFiles-fm(0x5d00d0001, 0x3300000062, 0xd00000022, 0xb10000000d, 0xffffffff00000000, 0xb1, 0x100000000, 0xc4203a564c, 0x2f, 0x2f, ...)
    /go/src/github.com/vmware/vic/vendor/github.com/vmware/govmomi/toolbox/command.go:72 +0x90
github.com/vmware/vic/vendor/github.com/vmware/govmomi/toolbox.(*CommandServer).Dispatch(0xc420258cc0, 0xc4203a5607, 0x74, 0x74, 0x8b, 0x3, 0xc4203a55f0, 0xc420154dd0, 0x8b)
    /go/src/github.com/vmware/vic/vendor/github.com/vmware/govmomi/toolbox/command.go:166 +0x440
github.com/vmware/vic/vendor/github.com/vmware/govmomi/toolbox.(*CommandServer).Dispatch-fm(0xc4203a5606, 0x75, 0x75, 0x15, 0xc420154e70, 0xc4203a5501, 0x15, 0x8b)
    /go/src/github.com/vmware/vic/vendor/github.com/vmware/govmomi/toolbox/command.go:81 +0x48
github.com/vmware/vic/vendor/github.com/vmware/govmomi/toolbox.(*Service).Dispatch(0xc4201c3340, 0xc4203a55f0, 0x8b, 0x8b, 0x0, 0x0, 0xc0)
    /go/src/github.com/vmware/vic/vendor/github.com/vmware/govmomi/toolbox/service.go:236 +0x177
github.com/vmware/vic/vendor/github.com/vmware/govmomi/toolbox.(*Service).Start.func1(0xc4201c3340, 0xc42025b840)
    /go/src/github.com/vmware/vic/vendor/github.com/vmware/govmomi/toolbox/service.go:185 +0x2ce
created by github.com/vmware/vic/vendor/github.com/vmware/govmomi/toolbox.(*Service).Start
    /go/src/github.com/vmware/vic/vendor/github.com/vmware/govmomi/toolbox/service.go:193 +0xd8
```

--- a/toolbox/command.go
+++ b/toolbox/command.go
@@ -511,8 +511,11 @@ func (c *CommandServer) ListFiles(header vix.CommandRequestHeader, data []byte) 
 
 	offset := r.Body.Offset + uint64(r.Body.Index)
 	total := uint64(len(files)) - offset
-
-	files = files[offset:]
+	if int(offset) < len(files) {
+		files = files[offset:]
+	} else {
+		total = 0 // offset is not valid (open-vm-tools behaves the same in this case)
+	}
 
 	var remaining uint64
 

--- a/toolbox/command_test.go
+++ b/toolbox/command_test.go
@@ -821,6 +821,10 @@ func TestVixFiles(t *testing.T) {
 		t.Errorf("expected %d, got %d", max, total)
 	}
 
+	// Test invalid offset, making sure it doesn't cause panic (issue #934)
+	ls.Body.Offset += 10
+	_ = c.Request(vix.CommandListFiles, ls)
+
 	// mv $dir ${dir}-old
 	mv = &vix.RenameFileRequest{
 		OldPathName: dir,

